File: DEMO/DEMO/ViewController.m
//
//  ViewController.m
//  DEMO
//
//  Created by Songwen Ding on 6/30/16.
//  Copyright © 2016 Songwen Ding. All rights reserved.
//

#import "ViewController.h"
#import "DEMO-Swift.h"

@interface ViewController ()
@end

@implementation ViewController {
    UIImageView *imageView;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    imageView = [[UIImageView alloc] init];
    [self.view addSubview:imageView];
    [PDFImage instance].useRamCache = YES;
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];
    
    imageView.frame = self.view.bounds;
    [[PDFImage instance] asyncGetImageWithResource:@"Group" bundle:[NSBundle mainBundle] page:1 size:imageView.bounds.size mainQueueBlock:^(UIImage *image) {
        imageView.image = image;
    }];
    for (int i = 0; i < 1000; i++) {
        id image = [[PDFImage instance] imageWithResource:@"Group" bundle:[NSBundle mainBundle] page:1 size:imageView.bounds.size];
        NSLog(@"%@",image);
    }
    
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: DEMO-Swift/DEMO/ViewController.swift
//
//  ViewController.swift
//  DEMO
//
//  Created by Songwen Ding on 6/30/16.
//  Copyright © 2016 Songwen Ding. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var imageView = UIImageView()
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(self.imageView)
        
        CCPDFImage.instance.useRamCache = true
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        self.imageView.frame = self.view.bounds
        CCPDFImage.instance.asyncGetImage("Group", bundle: NSBundle.mainBundle(), page: 1, size: self.imageView.bounds.size) { [weak self](image) in
            self?.imageView.image = image
        }
        //self.imageView.image = CCPDFImage.instance.image("Group", bundle: NSBundle.mainBundle(), page: 1, size: self.imageView.bounds.size)
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }


}

